### PR TITLE
More typescript conversions

### PR DESCRIPTION
**Pull request by CameronDiver**

### CODE REVIEW

File: src/lib/constants.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const supervisorNetworkInterface = 'supervisor0';
+
+export = {
+	rootMountPoint,
+	databasePath: checkString(process.env.DATABASE_PATH) || '/data/database.sqlite',
+	dockerSocket: process.env.DOCKER_SOCKET || '/var/run/docker.sock',
+	supervisorImage: checkString(process.env.SUPERVISOR_IMAGE) || 'resin/rpi-supervisor',
+	ledFile: checkString(process.env.LED_FILE) || '/sys/class/leds/led0/brightness',
+	// Only used in development
+	forceSecret: {
```
Comment (pcarranzav) [inline src/lib/constants.ts]: This can be removed (it stopped being used with #482)

File: src/lib/constants.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	configJsonPathOnHost: checkString(process.env.CONFIG_JSON_PATH),
+	proxyvisorHookReceiver:
+		checkString(process.env.RESIN_PROXYVISOR_HOOK_RECEIVER) || 'http://0.0.0.0:1337',
+	apiEndpointFromEnv: checkString(process.env.API_ENDPOINT),
+	configJsonNonAtomicPath: '/boot/config.json',
+	defaultPubnubSubscribeKey: process.env.DEFAULT_PUBNUB_SUBSCRIBE_KEY,
+	defaultPubnubPublishKey: process.env.DEFAULT_PUBNUB_PUBLISH_KEY,
+	defaultMixpanelToken: process.env.DEFAULT_MIXPANEL_TOKEN,
+	supervisorNetworkInterface: supervisorNetworkInterface,
+	allowedInterfaces: [ 'resin-vpn', 'tun0', 'docker0', 'lo', supervisorNetworkInterface ],
```
Comment (pcarranzav) [inline src/lib/constants.ts]: I just realized that this is outdated,we should probably think about what interfaces the supervisor listens on... I think it might only need to be resin-vpn, lo, and the supervisorNetworkInterface. But it might be better left for a separate PR.

File: src/lib/constants.ts (hunk 1 above)
Comment (CameronDiver) [reply src/lib/constants.ts]: Sure, will do it in this PR. In general I'm trying to change as little as possible, but given this is an easy win I can do that.

File: src/lib/constants.ts (hunk 2 above)
Comment (CameronDiver) [reply src/lib/constants.ts]: Yeah I'll save it for another PR, making an issue now.

Comment (CameronDiver) [reply src/lib/constants.ts]: https://github.com/resin-io/resin-supervisor/issues/639

File: src/lib/constants.coffee [hunk 3]
```diff
@@ -1,43 +0,0 @@
...
-	allowedInterfaces: ['resin-vpn', 'tun0', 'docker0', 'lo', supervisorNetworkInterface]
-	appsJsonPath: process.env.APPS_JSON_PATH ? '/boot/apps.json'
-	ipAddressUpdateInterval: 30 * 1000
-	imageCleanupErrorIgnoreTimeout: 3600 * 1000
-	maxDeltaDownloads: 3
-	defaultVolumeLabels: {
-		'io.resin.supervised': 'true'
-	}
-
-process.env.DOCKER_HOST ?= "unix://#{module.exports.dockerSocket}"
```
Comment (pcarranzav) [inline src/lib/constants.coffee]: @CameronDiver this is missing in the ts file and I think it's the reason you're seeing errors

File: src/lib/validation.ts [hunk 4]
```diff
@@ -1,28 +1,24 @@
...
-}
-
-export interface LabelObject {
-	[name: string]: string;
-}
-
 const ENV_VAR_KEY_REGEX = /^[a-zA-Z_][a-zA-Z0-9_]*$/;
 const LABEL_NAME_REGEX = /^[a-zA-Z][a-zA-Z0-9\.\-]*$/;
 
+type NullableString = string | undefined | null;
```
Comment (pcarranzav) [inline src/lib/validation.ts]: Should this also go in types.ts?

File: test/lib/conversions.spec.coffee [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+conversion = require '../../src/lib/conversions'
+
+describe 'conversions', ->
+
+	describe 'envArrayToObject', ->
+		it 'should convert an env array to an object', ->
+			expect(conversion.envArrayToObject([
+				'key=value'
+				'test1=test2'
+				'k=v'
```
Comment (pcarranzav) [inline test/lib/conversions.spec.coffee]: It would be good to test with a value that has an `=`

File: src/lib/validation.ts (hunk 4 above)
Comment (CameronDiver) [reply src/lib/validation.ts]: I'm not sure, I've tried to stick to only moving types that will be used in another place, whereas this was simply to avoid having to write out the value at every place. If another module finds itself needing it, I'll definitely move it out.
